src/osp/Universe.cpp

```diff
@@ -100,7 +95,15 @@ Vector3 Universe::sat_calc_pos_meters(Satellite referenceFrame, Satellite target
     return Vector3(sat_calc_pos(referenceFrame, target)) / gc_units_per_meter;
 }
 
-void Universe::update_sat_coordspace(uint32_t coordSpace)
+std::pair<coordspace_index_t, CoordinateSpace&> Universe::coordspace_create()
+{
+    coordspace_index_t const index = m_coordSpaces.size();
+    return {index, *m_coordSpaces.emplace_back(CoordinateSpace{})};
+
+    // TODO: find deleted spaces instead of emplacing back each time
```

Comment:
Maybe the function that removes items should compact the items in the vector, so that after removal, all the empty spots are at the end?

Reply:
The the element indices have to be stable however, so compacting cannot be done.